Refactor IRVestelAC class to use portable code.

crankyoldgit opened the pull request:
* Replace bit packed code types and use portable code instead.
  - Sad, because the bit packed code was much cleaner & easier to read.
  - This addresses some compiler warning/note messages.
* Add some more class methods for timers.
* Add the footer signature to the constructed timer messages.

Passes all the existing unit tests so it should be fine.

FYI @EUA

crankyoldgit committed:
Refactor IRVestelAC class to use portable code.
* Replace bit packed code types and use portable code instead.
  - Sad, because the bit packed code was much cleaner & easier to read.
  - This addresses some compiler warning/note messages.
* Add some more class methods for timers.
* Add the footer signature to the constructed timer messages.

Passes all the existing unit tests so it should be fine.

FYI @EUA

markszabo commented on src/ir_Vestel.h:
```diff
@@ -54,8 +54,23 @@ const uint8_t kVestelACTurbo = 7;
+const uint8_t kVestelACTimerFlagOffset = kVestelACHourOffset + 7;
+const uint8_t kVestelACMinuteOffset = 44;
+
+/*
```
Do we need to keep this here as comment?

crankyoldgit replied:
Yeah, I was lazy. I've removed it and added a detailed comment describing the bit-wise breakdown of the message.

Thanks for the good feedback.

crankyoldgit committed:
Remove old structure and add useful comments.
- Changes based on code review feedback.
- Remove un-needed stateReset() invocation.
